File: BackendKaikan/src/main/java/com/kaikan/model/pedido/Pedido.java
package com.kaikan.model.pedido;


import com.kaikan.generics.BaseEntity;
import com.kaikan.model.detallePedido.DetallePedido;
import com.kaikan.model.usuario.Usuario;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "pedido")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Pedido extends BaseEntity {
    @ManyToOne
    @JoinColumn(name = "id_usuario")
    @NotNull
    private Usuario usuario;
    @NotNull
    private LocalDateTime fecha_pedido;
    private LocalDateTime fecha_pago;
    @Enumerated(EnumType.STRING)
    private Estado estado;
    @OneToMany(mappedBy = "pedido", cascade = CascadeType.ALL)
    @NotEmpty
    private List<DetallePedido> detallesPedido;


}

File: Frontend-Kaikan-Restobar/src/app/core/models/pedido.ts
import { Usuario } from "./usuario";

export class Pedido{
  id:number|null;
  estado:string;
  fecha_pago:string;
  fecha_pedido:string;
  usuario:Usuario;

  constructor(id:number|null,estado:string,fecha_pago:string,
    fecha_pedido:string,usuario:Usuario
  ){
    this.id=id;
    this.estado=estado;
    this.fecha_pago=fecha_pago;
    this.fecha_pedido=fecha_pedido;
    this.usuario=usuario
  }
}
